storage/pebble/blocks.go

```diff
@@ -270,5 +276,26 @@ func (b *Blocks) getBlock(keyCode byte, key []byte) (*models.Block, error) {
 		return nil, fmt.Errorf("failed to get block: %w", err)
 	}
 
-	return models.NewBlockFromBytes(data)
+	block, err := models.NewBlockFromBytes(data)
+	if err != nil {
+		return nil, err
+	}
+
+	if b.chainID == flowGo.Testnet && slices.Contains(testnetBrokenParentHashHeights, block.Height) {
+		data, err := b.store.get(blockHeightKey, uint64Bytes(block.Height-1))
+		if err != nil {
+			return nil, fmt.Errorf("failed to get block: %w", err)
+		}
+		parentBlock, err := models.NewBlockFromBytes(data)
+		if err != nil {
+			return nil, err
+		}
+		parentBlockHash, err := parentBlock.Hash()
```

Comment:
can we add some comments about why we need to do this?

Reply:
Good point :+1: I have added a descriptive comment in: https://github.com/onflow/flow-evm-gateway/pull/566/commits/f14ad0cb903abc7270d6ebbacda6e014d5a030ef